jupyter_server_nbmodel/handlers.py

```diff
@@ -227,7 +228,11 @@ async def _execute_snippet(
                 del ycell["outputs"][:]
                 ycell["execution_count"] = None
                 ycell["execution_state"] = "running"
-
+                if metadata["record_timing"]:
+                    time_info = ycell["metadata"].get("execution",{})
+                    time_info["start_time"] = datetime.now(timezone.utc).isoformat()[:-6]
```

Comment:
For the metadata I think we will want to match the exact identifiers as used on the frontend, so instead of `start_time` we would use `shell.execute_reply.started` and for end time - `shell.execute_reply` - although these are less readable, it would make `jupyter-server-nbmodel` a drop-in replacement and compatible with existing extensions like [`jupyterlab-execute-time`](https://github.com/deshaw/jupyterlab-execute-time).

There also should be `execution_failed` for when the execution fails.